app/src/androidUnitTest/kotlin/de/westnordost/streetcomplete/data/visiblequests/QuestsHiddenControllerTest.kt

```diff
@@ -39,27 +45,51 @@ class QuestsHiddenControllerTest {
         val q2 = osmQuestKey(elementId = 2)
         val q3 = OsmNoteQuestKey(3)
         val q4 = OsmNoteQuestKey(4)
+        val q5 = AtpQuestKey(5)
+        val q6 = AtpQuestKey(6)
         on(osmDb.getAll()).thenReturn(listOf(OsmQuestHiddenAt(q1, 123L)))
         on(notesDb.getAll()).thenReturn(listOf(NoteQuestHiddenAt(q3.noteId, 124L)))
         on(notesDb.getTimestamp(q4.noteId)).thenReturn(null)
+        on(atpDb.getAll()).thenReturn(listOf(AtpQuestHiddenAt(q5.atpEntryId, 125L)))
 
         assertEquals(ctrl.get(q1), 123L)
         assertNull(ctrl.get(q2))
         assertEquals(ctrl.get(q3), 124L)
         assertNull(ctrl.get(q4))
+        assertEquals(ctrl.get(q5), 125L)
+        assertNull(ctrl.get(q6))
     }
 
     @Test fun getAllNewerThan() {
         val h1 = OsmQuestHiddenAt(osmQuestKey(elementId = 1), 250)
         val h2 = OsmQuestHiddenAt(osmQuestKey(elementId = 2), 123)
         val h3 = NoteQuestHiddenAt(2L, 500)
         val h4 = NoteQuestHiddenAt(3L, 123)
+        val h5 = AtpQuestHiddenAt(4L, 23)
+        val h6 = AtpQuestHiddenAt(5L, 100000)
 
         on(osmDb.getAll()).thenReturn(listOf(h1, h2))
         on(notesDb.getAll()).thenReturn(listOf(h3, h4))
+        on(atpDb.getAll()).thenReturn(listOf(h5, h6))
 
+        assertEquals<Pair<QuestKey, Long>>(
+            Pair(AtpQuestKey(h6.allThePlacesEntryId), 100000),
+            ctrl.getAllNewerThan(123L)[0]
+        )
+        assertEquals<Pair<QuestKey, Long>>(
+            Pair(OsmNoteQuestKey(h3.noteId), 500L),
+            ctrl.getAllNewerThan(123L)[1]
+        )
+        assertEquals<Pair<QuestKey, Long>>(
+            Pair(h1.key, 250L),
+            ctrl.getAllNewerThan(123L)[2]
+        )
+        //TODO above works, below fails, why? Oh and below fails while claiming no difference
+        // assertEquals is called on listOf in few places so it does not appear due to comparing
+        // specific list instances instead of actual content
         assertEquals(
             listOf(
+                AtpQuestKey(h6.allThePlacesEntryId) to 100000,
```

Comment:
```suggestion
        assertEquals(
            listOf(
                AtpQuestKey(h6.allThePlacesEntryId) to 100000L,
```

I am pretty sure because `100000` is an `Integer`, so the resulting `List` will be of type `List<Pair<QuestKey, Any>>`. Lists can only be equal if the type is the same.